drivers/SmartThings/samsung-audio/src/disco.lua

```diff
@@ -120,20 +126,24 @@ function Disco.find(deviceid, callback)
       local headers = process_response(val)
       local ip, port = headers["location"]:match(
                              "http://([^,/]+):([^/]+)") -- TODO : We need to check the xml filename for samsung audio device for ex: http://192.168.0.1:59666/rootDesc.xml
-      local id = headers["usn"]
-
+      -- local id = headers["usn"]
+ 
       -- TODO how do I know the device that responded is actually a samsung-audio device
       -- potentially will need to make a request to the endpoint
       local meta = fetch_device_metadata(headers["location"])
       local speaker_name = "samsung-audio speaker"
       local speaker_model = "unknown samsung-audio"
+      local speaker_mac = "000000000000"
       if not meta then
           meta = {}
           log.trace("fetch_device_metadata INFO is NULL")
       else
           speaker_name = meta.name
           speaker_model = meta.model
+          speaker_mac = meta.mac
       end
+      local id = speaker_mac
```

Comment:
At this point we should also check to make sure `speaker_mac` isn't *still* `nil` and early `break` or early `return`, no sense in continuing if the MAC isn't valid.

Reply:
Done